### PR TITLE
Added 'Offline Maps' to Navigation Drawer

### PR DESCRIPTION
**Pull request by jtj9817**
## Description
Starting point for the features requested in #534 . 

## Changes
- Created 'Offline Maps' icon
-  Created ' case R.id.nav_offlinemaps:' switch-statement condition on MainACtivity.java
- 'fragment_offlinemaps.xml' created
- Added '<string name="navigation_drawer_offlinemaps">Offline Maps</string>' to strings.xml Created 'OfflineMapsFragment.java' class under io.github.project_travel_mate/utilities folder

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`
- [x] Ran the app on Genymotion Emulator with Google Pixel 2 (API 26) and no crashes.

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

# Screenshot

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      <window_info id="Favorites" order="2" side_tool="true" />
+      <window_info anchor="bottom" id="Debug" order="3" weight="0.4" />
+      <window_info anchor="right" content_ui="combo" id="Hierarchy" order="2" weight="0.25" />
+      <window_info anchor="bottom" id="Inspection" order="5" weight="0.4" />
+      <window_info anchor="right" id="Commander" order="0" weight="0.4" />
+      <window_info anchor="right" id="Ant Build" order="1" weight="0.25" />
+      <window_info anchor="bottom" id="Message" order="0" />
+      <window_info anchor="bottom" id="Cvs" order="4" weight="0.25" />
+    </layout>
+  </component>
```
Comment (Swati4star) [inline .idea/workspace.xml]: remove .idea folder from commit

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java [hunk 2]
```diff
@@ -233,7 +233,9 @@ private Fragment getFragmentByNavMenuItemId(int id) {
             case R.id.nav_utility:
                 fragment = UtilitiesFragment.newInstance();
                 break;
-
+                //Work In Progress Offline Maps
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: is it work in progress or done?

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java [hunk 3]
```diff
@@ -182,7 +182,7 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
             mCityAdapter.updateData(mCities);
 
-            if (mSpotlightShownCount <= 3) {
+            if (mSpotlightShownCount <= 4) {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: why?

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 2 above)
Comment (jtj9817) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: Still a work in progress for the MainActivity.java file as I'm still trying to figure out how the feature should be implemented.

File: Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java (hunk 3 above)
Comment (jtj9817) [reply Android/app/src/main/java/io/github/project_travel_mate/destinations/CityFragment.java]: My bad, this was an experimentation I was doing on trying to solve the #539 issue.

File: .idea/workspace.xml (hunk 1 above)
Comment (jtj9817) [reply .idea/workspace.xml]: Will do!

**Comment by Swati4star** [comment]
@jtj9817 You have opened a number of PRs. Let's start merging them one by one. 
Starting with this one, This one still has .idea folder, .gitignore also 

and is it still a work in progress? If yes, lets be done with this. (either close it or make it ready to be merged)

File: Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java (hunk 2 above)
Comment (jtj9817) [reply Android/app/src/main/java/io/github/project_travel_mate/MainActivity.java]: @Swati4star Hey, just cleaned up this branch for the merging process. Please review it again, thanks!

**Comment by Swati4star** [comment]
@jtj9817 Are you going to clean it up?

**Comment by jtj9817** [comment]
Sorry @Swati4star for the late reply but yes I'm going to clean this one up, just been busy with school for this past week.

File: Android/app/src/main/res/layout/city_history_listitem.xml [hunk 4]
```diff
@@ -2,23 +2,26 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
-    android:orientation="vertical"
     android:layout_margin="5dp"
-    android:paddingStart="5dp"
-    android:paddingEnd="5dp">
+    android:orientation="vertical"
```
[the remark this answers is not in the document]
Comment (jtj9817) [reply Android/app/src/main/res/layout/city_history_listitem.xml]: Seems that this change was related to another bug/issue I was trying to fix. I reverted this change. Also, the .idea folder and .gitignore file are no longer in this branch as I see it from my end as seen here - https://github.com/jtj9817/Travel-Mate . Maybe I missed it on some other subfolder?

**Comment by Swati4star** [comment]
Instead of png, you should use an xml drawable file

**Comment by jtj9817** [comment]
> Instead of png, you should use an xml drawable file

Will try to convert the PNG to a vector asset. Probably a day.

**Comment by jtj9817** [comment]
@Swati4star Just made the recent suggestion that you requested. I hope it's good for a merge now, thanks!

**Pull request closed without merging**